### PR TITLE
feat: enhance `useEditorDecorations`

### PR DESCRIPTION
**Pull request by antfu**
Currently `useEditorDecorations` does not retrigger when the content of the document changes

- Expose `trigger` function for users to manually trigger it for cases where they need manual triggers
- Introduce options, with `triggersOn`
- By default triggers on document changes
- Expose `editor` to the getter function

### CODE REVIEW

**Review by KermanX — APPROVED** [review]
Thanks a lot!

I renamed `trigger` to `update`, which I think is more intuitive, but I am not sure if `trigger` is a convention.

Also, there is a potential problem that `effect` and `documentChanged` will together trigger the update twice.

**Comment by antfu** [comment]
> there is a potential problem that effect and documentChanged may together trigger the update twice.

Yeah, that's why we need debounce and throttle.

File: packages/core/src/composables/useEditorDecorations.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+import { useDisposable } from './useDisposable'
+import { useDocumentText } from './useDocumentText'
+
+export interface UseEditorDecorationsOptions {
+  /**
+   * The triggers to update the decorations.
+   *
+   * @default ['effect', 'documentChanged']
+   */
+  triggersOn?: ('effect' | 'documentChanged' | WatchSource)[]
```
Comment (antfu) [inline packages/core/src/composables/useEditorDecorations.ts:15]: If `trigger` has been changed to `update`, should this also be changed? Like `events`?

**Comment by KermanX** [comment]
I renamed `triggersOn` to `updateOn`, and removed the usage of passing `WatchSource[]`. Now the update function won't be triggered twice. About the throttle and debounce options, I think they won't be often used because `workspace.onDidChangeTextDocument` seems to have been throttled. Also, throttle and debounce has been provided `@reactive-vscode/vueuse`.

**Pull request merged**